Assets/Scripts/Models/Buildable/Components/TemperatureConnection.cs

```diff
@@ -0,0 +1,188 @@
+    public class TemperatureComponent : BuildableComponent
+    {
+        public TemperatureComponent()
+        {
+        }
+
+        private TemperatureComponent(TemperatureComponent other) : base(other)
+        {
+            Requires = other.Requires;
+            Outputs = other.Outputs;
+            ParamsDefinitions = other.ParamsDefinitions;
+        }
+
+        [XmlElement("Requires")]
+        [JsonProperty("Requires")]
+        public InputInfo Requires { get; set; }
+
+        [XmlElement("Outputs")]
+        [JsonProperty("Outputs")]
+        public OutputInfo Outputs { get; set; }
+
+        [XmlElement("ParameterDefinitions")]
+        [JsonProperty("ParameterDefinitions")]
+        public TemperatureConnectionParameterDefinitions ParamsDefinitions { get; set; }
+
+        public override bool RequiresSlowUpdate
+        {
+            get
+            {
+                return true;
+            }
+        }
+
+        public float OutputRate
+        {
+            get
+            {
+                return FurnitureParams[ParamsDefinitions.OutputRate.ParameterName].ToFloat();
+            }
+
+            set
+            {
+                FurnitureParams[ParamsDefinitions.OutputRate.ParameterName].SetValue(value);
+            }
+        }
+
+        public override BuildableComponent Clone()
+        {
+            return new TemperatureComponent(this);
+        }
+
+        public override bool CanFunction()
+        {
+            bool canFunction = true;
+
+            // check if all requirements are fullfilled
+            if (Requires != null)
+            {
+                if (ParentFurniture.Tile.TemperatureUnit.TemperatureInKelvin < Requires.MinLimit || ParentFurniture.Tile.TemperatureUnit.TemperatureInKelvin > Requires.MaxLimit)
+                {
+                    canFunction = false;
+                }
+            }
+
+            return canFunction;
+        }
+
+        public override void FixedFrequencyUpdate(float deltaTime)
+        {
+            if (Outputs != null && OutputRate != 0)
+            {
+                if (Outputs.MaxLimit != -1)
+                {
+                    if (ParentFurniture.Tile.Room != null && ParentFurniture.Tile.TemperatureUnit.TemperatureInKelvin > Outputs.MaxLimit)
+                    {
+                        return;
+                    }
+                }
+
+                if (Outputs.MinLimit != -1)
+                {
+                    if (ParentFurniture.Tile.Room != null && ParentFurniture.Tile.TemperatureUnit.TemperatureInKelvin < Outputs.MinLimit)
+                    {
+                        return;
+                    }
+                }
+
+                // Else we are fine :D
+                World.Current.temperature.ProduceTemperatureAtFurniture(ParentFurniture, OutputRate, deltaTime);
+            }
+        }
+
+        protected override void Initialize()
+        {
+            componentRequirements = Requirements.Temperature;
```

Comment:
Will it matter...?  Because it won't ever tell the system to display the icon unless `Requires != null`.  I was just following the pattern of every other component where they would set that requirement, though this may be  a bridge case?